service/controller/resource/values/create.go

```diff
@@ -0,0 +1,68 @@
+package values
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/microerror"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/giantswarm/config-controller/pkg/generator/key"
+	controllerkey "github.com/giantswarm/config-controller/service/controller/key"
+)
+
+func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	app, err := controllerkey.ToAppCR(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	configVersion, ok := app.GetAnnotations()[key.ConfigVersion]
+	if !ok {
+		r.logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("App CR %q is missing %q annotation", app.Name, key.ConfigVersion))
+		return nil
+	}
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("generating app %#q config version %#q", app.Spec.Name, configVersion))
+	configmap, secret, err := r.generateConfig(ctx, r.installation, app.Namespace, app.Spec.Name, configVersion)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("generated app %#q config version %#q", app.Spec.Name, configVersion))
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating configmap %s/%s", configmap.Namespace, configmap.Name))
+	err = r.k8sClient.CtrlClient().Create(ctx, configmap)
+	if apierrors.IsAlreadyExists(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("configmap %s/%s already exists", configmap.Namespace, configmap.Name))
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created configmap %s/%s", configmap.Namespace, configmap.Name))
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("creating secret %s/%s", secret.Namespace, secret.Name))
+	err = r.k8sClient.CtrlClient().Create(ctx, secret)
+	if apierrors.IsAlreadyExists(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("secret %s/%s already exists", secret.Namespace, secret.Name))
+	} else if err != nil {
+		return microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("created secret %s/%s", secret.Namespace, secret.Name))
```

Comment:
We need to update CM/Secret as well.

Reply:
Do you mean running `r.k8sClient.CtrlClient().Update(ctx, configmap)` if CM/Secret already exists?

Reply:
@kopiczko ^ :D

Reply:
Yes